webui/react/src/hooks/useModal/Model/useModalModelCreate.tsx

```diff
@@ -218,19 +267,32 @@ const useModalModelCreate = ({ onClose }: Props = {}): ModalHooks => {
 
   const getModalProps = useCallback(
     (state: ModalState): Partial<ModalFuncProps> => {
+      const disableWorkspaceModelCreation = isWorkspace
+        ? !canCreateModelWorkspace(workspaceId)
+        : false;
       return {
         className: css.base,
         closable: true,
         content: getModalContent(state),
         icon: null,
         maskClosable: true,
-        okButtonProps: { disabled: !modelName, form: FORM_ID, htmlType: 'submit' },
-        okText: 'Create Model',
+        okButtonProps: {
+          disabled: !modelName || (isWorkspace && disableWorkspaceModelCreation),
```

Comment:
Can this be simplified to 

```
!modelName || disableWorkspaceModelCreation
```

since if `isWorkspace` is `false` then `disableWorkspaceModelCreation` has to be false too.